Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/UserDataImportWarningDialog.kt

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.dialogs
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.core.net.toUri
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.ViewModelProvider
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.dolphinemu.dolphinemu.R
+import org.dolphinemu.dolphinemu.activities.UserDataActivity
+import org.dolphinemu.dolphinemu.model.UserDataViewModel
+import org.dolphinemu.dolphinemu.utils.ThreadUtil
+import kotlin.system.exitProcess
+
+class UserDataImportWarningDialog : DialogFragment() {
+    private lateinit var viewModel: UserDataViewModel
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        viewModel = ViewModelProvider(requireActivity())[UserDataViewModel::class.java]
+
+        val dialog = MaterialAlertDialogBuilder(requireContext())
+            .setMessage(R.string.user_data_import_warning)
+            .setNegativeButton(R.string.no) { dialog: DialogInterface, _: Int -> dialog.dismiss() }
+            .setPositiveButton(R.string.yes) { dialog: DialogInterface, _: Int ->
+                dialog.dismiss()
+
+                ThreadUtil.runOnThreadAndShowResult(
+                    requireActivity(),
+                    R.string.import_in_progress,
+                    R.string.do_not_close_app,
+                    {
+                        resources.getString(
+                            (activity as UserDataActivity).importUserData(
+                                requireArguments().getString(KEY_URI_RESULT)!!.toUri()
+                            )
+                        )
+                    }) {
+                    if (viewModel.mustRestartApp.value!!) {
+                        exitProcess(0)
+                    }
+                }
+            }
+        return dialog.create()
+    }
+
+    companion object {
+        const val TAG = "UserDataImportWarningDialog"
+        const val KEY_URI_RESULT = "uri_result"
```

Comment:
In the context of this dialog, it would make more sense to name the key something that doesn't include the word "result".